src/v2/BaseLoginRouter.js

```diff
@@ -84,12 +85,21 @@ export default Router.extend({
     // Hide until unitial render
     this.hide();
 
-    configIdxJsClient(this.appState);
+    this.listenTo(this.appState, 'change:deviceFingerprint', this.updateDeviceFingerprint);
+    this.listenTo(this.appState, 'error', this.handleError);
     this.listenTo(this.appState, 'updateAppState', this.handleUpdateAppState);
     this.listenTo(this.appState, 'remediationError', this.handleIdxResponseFailure);
     this.listenTo(this.appState, 'restartLoginFlow', this.restartLoginFlow);
   },
 
+  updateDeviceFingerprint() {
+    const authClient = this.settings.getAuthClient();
+    const fingerprint = this.appState.get('deviceFingerprint');
+    if (fingerprint) {
+      authClient.http.setRequestHeader('X-Device-Fingerprint', fingerprint);
```

Comment:
IIRC, we choose interceptor approach was because `X-Device-Fingerprint` only being set whenever there is value in `appState`.
With this approach, will the `X-Device-Fingerprint` header be clear when `appState.deviceFingerprint` has been clear.

Reply:
The header will not be cleared. This is a change in behavior, and the test is updated. I discussed with @jmelberg-okta and seems there should not be any negative effect to keeping the header set.